src/realm/object-store/sync/app.hpp

```diff
@@ -245,9 +245,21 @@ class App : public std::enable_shared_from_this<App>,
         SharedApp m_parent;
     };
 
+    /// Retrieve a cached app instance if one was previously generated for `config`'s app_id+base_url combo,
+    /// otherwise generate and return a new instance and persist it in the cache.
     static SharedApp get_shared_app(const Config& config, const SyncClientConfig& sync_client_config);
+
+    /// Generate and return a new app instance for the given config, bypassing the app cache.
     static SharedApp get_uncached_app(const Config& config, const SyncClientConfig& sync_client_config);
-    static std::shared_ptr<App> get_cached_app(const std::string& app_id);
+
+    /// Return a cached app instance if one was previously generated for the `app_id`+`base_url` combo using
+    /// `get_shared_app`.
+    /// If base_url is not provided, and there are multiple cached apps with the same app_id but different base_urls,
+    /// then a random one will be returned.
```

Comment:
It's not really a random one - it's the first one that get_shared_app was called with - it's non-deterministic, but not random.

Reply:
oh, i thought iteration on `unordered_map` is deterministic, but not necessarily in the order of insertion?

Reply:
I was just saying the comment isn't quite right. It's not a random one that will get returned - it's just undefined.

Reply:
ah okay i see what you mean now. will update the docstring 👍🏼